delivery/src/main/java/com/storyous/delivery/common/AutodeclineCountdown.kt

```diff
@@ -0,0 +1,62 @@
+package com.storyous.delivery.common
+
+import android.os.CountDownTimer
+import android.widget.TextView
+import androidx.constraintlayout.widget.Group
+import androidx.core.view.isVisible
+import com.storyous.commonutils.TimestampUtil
+import com.storyous.delivery.common.api.DeliveryOrder
+import kotlin.math.floor
+
+class AutodeclineCountdown(
+    private val textView: TextView,
+    timeToDecline: Long
+) : CountDownTimer(timeToDecline, SECOND_MILLIS) {
+
+    companion object {
+        const val TIME_UNIT = 60f
+        const val SECOND_MILLIS = 1000L
+
+        fun init(deliveryOrder: DeliveryOrder, visibilityGroup: Group, countdownText: TextView): AutodeclineCountdown? {
+            val visible = deliveryOrder.state == DeliveryOrder.STATE_NEW &&
+                deliveryOrder.timing?.autoDeclineAfter != null &&
+                deliveryOrder.timing.autoDeclineAfter > TimestampUtil.getDate()
+            visibilityGroup.isVisible = visible
+            return if (visible) {
+                AutodeclineCountdown(
+                    countdownText,
+                    TimestampUtil.getTimeAgo(deliveryOrder.timing!!.autoDeclineAfter!!.time)
+                ).apply { start() }
+            } else {
+                null
+            }
+        }
+    }
+
+    override fun onTick(millisUntilFinished: Long) {
+        textView.text = getRemainingTime(millisUntilFinished / SECOND_MILLIS)

```

Comment:
Nejde použít prostě jen:
```
textView.text = DateUtils.HMS.format(Date(millisUntilFinished))
```

Reply:
Chcem, aby to nebylo dlouhý a nějak rozlišit zobrazení hodina:minuta a minuta:sekunda. To HMS úplně nedělá.

Reply:
Tak pak stačí jen oříznutí hodnoty 10h:20m:30s na |10h:20m|:30s nebo 10h:|20m:30s|